### ISSUE TITLE
AttributeError: 'Series' object has no attribute 'dtypes'

### ISSUE THREAD

**Issue by lancelot1969** [pyviz/datashader#617]
Running census datashader example on Windows10:

```
cvs = ds.Canvas(plot_width, plot_height, *webm(*USA))
agg = cvs.points(df, 'easting', 'northing'
```

Getting error:

```
---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
<timed exec> in <module>()

~\AppData\Local\Continuum\anaconda3\lib\site-packages\datashader\core.py in points(self, source, x, y, agg)
    158         if agg is None:
    159             agg = count_rdn()
--> 160         return bypixel(source, self, Point(x, y), agg)
    161 
    162     def line(self, source, x, y, agg=None):

~\AppData\Local\Continuum\anaconda3\lib\site-packages\datashader\core.py in bypixel(source, canvas, glyph, agg)
    508     # by only retaining the necessary columns:
    509     # https://github.com/bokeh/datashader/issues/396
--> 510     if categorical_in_dtypes(source.dtypes.values):
    511         # Preserve column ordering without duplicates
    512         cols_to_keep = OrderedDict({col: False for col in source.columns})

AttributeError: 'Series' object has no attribute 'dtypes'
```

**Comment by jbednar** [comment]
Sounds like you have an incompatible combination of library versions.  What versions of datashader, pandas, and dask are you using?

**Comment by philippjfr** [comment]
Are you sure that the ``df`` in our example is not a Series?

**Comment by jbednar** [comment]
Not sure what you mean; in census.ipynb it's a Dask DataFrame.

**Comment by philippjfr** [comment]
I assumed something was modified because if you follow the code the only way I can see you can end up with that error if the ``df`` in:

```
agg = cvs.points(df, ...
```

is actually a ``Series``.

**Comment by lancelot1969** [comment]
Thanks. Resolved an issue by downgrading dask from 0.18.1 to 0.17.4

**Comment by vahidtz** [comment]
dtypes has changed to dtype in dask.series data collection.
